src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ScoreSystem.java

```diff
@@ -89,13 +98,19 @@ public void onActivate(ActivateEvent event, EntityRef entity) {
 
```

Comment:
Let's move the entire logic here inside `onActivate` to a method called- `checkAndResetGameOnScore`

This would do three things, have one method for each- 
1. Check if any team did score. (Check for the team in base's LASTeamComponent and for the heldItem to have the opposite team's FlagComponent)
2. Increment score. 
3. Reset the round- (This should be the same method for both teams and should completely reset the round however we want)